src/time_unit/mod.rs

```diff
@@ -156,6 +156,20 @@ pub trait TimeUnitSpec {
     /// assert_eq!(2, schedule.days_of_month().count());
     /// ```
     fn count(&self) -> u32;
+
+    /// Checks if this TimeUnitSpec is specified (thus not created with a *)
+    /// # Example
+    /// ```
+    /// use cron::{Schedule,TimeUnitSpec};
+    /// use std::str::FromStr;
+    ///
+    /// let expression = "* * * 1,15 * * *";
+    /// let schedule = Schedule::from_str(expression).expect("Failed to parse expression.");
+    ///
+    /// assert_eq!(true, schedule.days_of_month().is_specified());
+    /// assert_eq!(false, schedule.months().is_specified());
+    /// ```
+    fn is_specified(&self) -> bool;
```

Comment:
I have a minor quibble with the name of this function: the field _is_ specified; the user had to write `*`. How would you feel about `is_all` instead, which would return the inverse?

Reply:
I'm a bit afraid that this implementation would break when you get to years though. If I would type Years.is_all() on 1970-2070 it should actually always return false right? Or maybe the end is actually nigh :p 

Yes you can specify 1-7 for weekdays, but then you actually did specify it, as being all weekdays.

My use case was (basically the reason why I came about this project) to change cron strings into launchd files. When you don't specify a year in a launchd file, it means that you want something to happen regardless of the year. I didn't want to print 100 or so years in my launchd file whenever I parsed a string with a * in it.

Reply:
> I'm a bit afraid that this implementation would break when you get to years though. If I would type Years.is_all() on 1970-2070 it should actually always return false right? Or maybe the end is actually nigh :p

Ha, good point. The impl I laid out would be the default impl, but I think we could easily justify overriding it in the impl for `Years` to just look at `is_some()`.

> Yes you can specify 1-7 for weekdays, but then you actually did specify it, as being all weekdays.

I see the distinction you're making, but I'm not certain I want `Schedule` to make that distinction. There's not currently a distinction between `*` and `?` (for fields that accept it), and intuitively I would expect `1-7` (or whichever inclusive range) to be treated the same for fields that have a fixed range (all of them but `Years`).

> My use case was (basically the reason why I came about this project) to change cron strings into launchd files. When you don't specify a year in a launchd file, it means that you want something to happen regardless of the year. I didn't want to print 100 or so years in my launchd file whenever I parsed a string with a * in it.

Does the default impl override for `Years` address this for you?

To be clear, I could be persuaded to have an `is_specified` method, it just feels a bit hazy semantically. I wanted to see if we could get by with something that wasn't strictly bound to the input expression syntax first.